libs/expo/shared/ui-components/src/lib/LibraryModal/index.tsx

```diff
@@ -0,0 +1,166 @@
+import Modal from 'react-native-modal';
+import TextBold from '../TextBold';
+import TextRegular from '../TextRegular';
+import LibCamera from './Camera';
+
+interface ILibraryModalProps {
+  onCapture: (file: ReactNativeFile) => void;
+  setModalVisible: (isModalVisible: boolean) => void;
+  isModalVisible: boolean;
+  setFiles: (files: ReactNativeFile[]) => void;
+  allowMultiple?: boolean;
+}
+
+export default function LibraryModal(props: ILibraryModalProps) {
+  const {
+    onCapture,
+    setModalVisible,
+    isModalVisible,
+    setFiles,
+    allowMultiple = true,
+  } = props;
+  const [isCameraOpen, setIsCameraOpen] = useState(false);
+
+  const [permission, requestPermission] = useCameraPermissions();
+
+  const closeModal = () => {
+    setModalVisible(false);
+  };
+
+  const getPermissionsAndOpenCamera = async () => {
+    if (permission) {
+      const { granted } = await requestPermission();
+      if (granted) {
+        setIsCameraOpen(true);
+      } else {
+        Alert.alert(
+          'Permission Denied',
+          'You need to grant camera permission to use this app'
+        );
+      }
+    }
+  };
+
+  const pickImage = async () => {
+    try {
+      const result = await ImagePicker.launchImageLibraryAsync({
+        mediaTypes: ImagePicker.MediaTypeOptions.Images,
+        allowsEditing: false,
+        allowsMultipleSelection: allowMultiple,
+      });
+
+      if (!result.canceled && result.assets) {
+        const uploadPromises = result.assets.map(async (asset) => {
+          const resizedPhoto = await resizeImage({ uri: asset.uri });
+          const file = new ReactNativeFile({
+            uri: resizedPhoto.uri,
+            name: asset?.fileName || Date.now().toString(),
+            type: asset.mimeType || 'image/jpeg',
+          });
+
+          return file;
+        });
+
+        const uploadedImages = await Promise.all(uploadPromises);
+
+        setFiles(uploadedImages);
+        setModalVisible(false);
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  return (
+    <Modal
+      animationIn="slideInUp"
+      animationOut="slideOutDown"
+      backdropOpacity={0.5}
+      onBackdropPress={closeModal}
+      isVisible={isModalVisible}
+      style={{ justifyContent: 'flex-end', margin: isCameraOpen ? 0 : 20 }}
+      useNativeDriverForBackdrop={true}
+    >
+      {isCameraOpen ? (
+        <LibCamera
+          setModalVisible={setModalVisible}
+          setIsCameraOpen={setIsCameraOpen}
+          onCapture={onCapture}
+        />
+      ) : (
+        <>
+          <View
+            style={{
+              backgroundColor: Colors.WHITE,
+              marginBottom: Spacings.xs,
+              borderRadius: 8,
+              shadowColor: Colors.BLACK,
+              shadowOffset: {
+                width: 0,
+                height: 2,
+              },
+              shadowOpacity: 0.25,
+              shadowRadius: 3.84,
```

Comment:
this is a very specific number. Can we use one of our existing constants here? We have an `xxs: 4`. I see this `3.84` in a few other places already in the code and I wonder if we can replace all of those, too

Reply:
3.84 is Material design specific elevation. It is very common usage. 
only that is why

Reply:
oh, I see. If it's that common, can we abstract it to a constant?

Reply:
what if I create a constant for a shadow? like an object. so we will just spread that everywhere shadow is needed

Reply:
created a constant object Shadow